### ISSUE TITLE
Is Provider.of(context, listen: false) equivalent of context.read()?

### ISSUE THREAD

**Issue by planets-9** [rrousselGit/provider#470]
Say I have this

    final model = Provider.of<Model>(context, listen: false); 
    final model = context.read<Model>(); 

Are they same or aren't they? If they are, then why I get the error when I use `read` inside `build()` method, `Provider.of()` works though. 

> Tried to use `context.read<Model>` inside either a `build` method or the `update` callback of a provider.

**Comment by rrousselGit** [comment]
For backward compatibility.

**Comment by planets-9** [comment]
@rrousselGit I am getting error.

**Comment by rrousselGit** [comment]
This is the expected behavior. See the documentation of read

**Comment by planets-9** [comment]
That means we shouldn't use 

```
final model = context.read<Model>(); 
```
in place of 
```
final model = Provider.of<Model>(context, listen: false); 
```

Correct?

**Comment by rrousselGit** [comment]
No, that means you are doing something you shouldn't inside `build`

**Comment by planets-9** [comment]
Got it, thank you so much, I shouldn't have watched the video alone, documentation is must when it comes to Provider. 

Thanks for the library.
